app/src/main/java/com/ece493/group5/adjustableaudio/models/HearingTestModel.java

```diff
@@ -0,0 +1,408 @@
+import android.util.Log;
+import android.widget.EditText;
+
+import com.ece493.group5.adjustableaudio.R;
+import com.ece493.group5.adjustableaudio.storage.Encrypter;
+import com.ece493.group5.adjustableaudio.storage.HearingTestResultListController;
+import com.ece493.group5.adjustableaudio.storage.Jsonizer;
+import com.ece493.group5.adjustableaudio.storage.SaveController;
+import com.ece493.group5.adjustableaudio.storage.Saver;
+import com.ece493.group5.adjustableaudio.views.HearingTestView;
+
+import java.io.UnsupportedEncodingException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.Observable;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+
+import androidx.appcompat.app.AlertDialog;
+
+public class HearingTestModel extends Observable
+{
+    private static final long BEEP_DURATION = 3000;
+    private static final String NUMBER_FREQUENCIES = "16";
+    private static final int MAX_DB = 100;
+    private static final int[] TONES = {30, 90, 233, 250, 347,
+                                        500, 907, 1000, 1353, 2000,
+                                        3533, 4000, 5267, 8000, 11333, 15667};
+    private static final double[] REFERENCE_FREQUENCY_DBHL_VALUES = {
+            60.0, 37.0, 19.0, 18.0,
+            14.6, 11.0, 6.0, 5.5,
+            5.5, 4.5, 6.5, 9.5,
+            14.8, 17.5, 23.0, 52.5};
+    private static final String DEFAULT_NAME = "Hearing Test";
+    public static final String PACKAGE_NAME = "com.ece493.group5.adjustableaudio";
+    private static final String LEFT_EAR = "L";
+    private static final String RIGHT_EAR = "R";
+    private static final String PAUSE_TAG = "pause";
+    private static final String UNPAUSE_TAG = "unpause";
+
+
+    private Context mContext;
+    private HearingTestView mView;
+
+    private ArrayList<ToneData> toneDataArrayList;
+    private String currentEar;
+    private String testName;
+    private Boolean testRunning;
+    private Boolean isPaused;
+    private Boolean maxVolumeReached;
+    private SoundPool soundPool;
+    private AudioManager audioManager;
+    private int currentSound;
+    private int progress;
+    private double dbHLLevel;
+    private float LVolume;
+    private float RVolume;
+    private ArrayList<Integer> soundPoolSounds;
+    private AudioManager.OnAudioFocusChangeListener onFocusChangeListener;
+
+
+    public HearingTestModel(Context mContext)
+    {
+        this.mContext = mContext;
+        this.currentEar = LEFT_EAR;
+        this.testRunning = false;
+        initTest();
+        initToneDataArrayList();
+        initSoundPool();
+        this.audioManager = (AudioManager) mContext.getSystemService(Context.AUDIO_SERVICE);
+        this.onFocusChangeListener = new AudioManager.OnAudioFocusChangeListener(){
```

Comment:
I don't think this is used. Why isn't this used? If it's not required, it probably should be removed.